ee/tabby-webserver/src/schema/email.rs

```diff
@@ -1,11 +1,39 @@
 use anyhow::Result;
 use async_trait::async_trait;
-use juniper::GraphQLObject;
+use juniper::{GraphQLEnum, GraphQLObject};
+use validator::Validate;
+
+#[derive(GraphQLEnum, Clone, Debug)]
+pub enum Encryption {
+    StartTls,
+    SslTls,
+    None,
+}
+
+#[derive(GraphQLEnum, Clone, Debug)]
+pub enum AuthMethod {
+    Plain,
+    Login,
+    XOAuth2,
+}
 
 #[derive(GraphQLObject)]
 pub struct EmailSetting {
     pub smtp_username: String,
     pub smtp_server: String,
+    pub from_address: String,
+    pub encryption: Encryption,
+    pub auth_method: AuthMethod,
+}
+
+#[derive(GraphQLObject, Validate)]
+pub struct EmailSettingInput {
+    #[validate(email)]
```

Comment:
smtp_username doesn't have to be email address,

example for aws:

<img width="1080" alt="image" src="https://github.com/TabbyML/tabby/assets/388154/0ed23f04-5161-450d-b730-114ccfc2d460">